### PR TITLE
Add IT for AWS OpenSearch using LocalStack & Testcontainers

### PR DESCRIPTION
**Pull request by eddumelendez**

### CODE REVIEW

File: langchain4j-opensearch/src/test/java/dev/langchain4j/store/embedding/opensearch/OpenSearchEmbeddingStoreAWSTest.java [hunk 1]
```diff
@@ -1,38 +1,64 @@
...
+@Testcontainers
 public class OpenSearchEmbeddingStoreAWSTest extends EmbeddingStoreIT {
 
-    /**
-     * To run the tests locally, you have to provide an Amazon OpenSearch domain. The code uses
-     * the credentials stored locally in your machine. For more information about how to configure
-     * your credentials locally, see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html.
-     */
+    @Container
+    private static final LocalStackContainer localstack = new LocalStackContainer(DockerImageName.parse("localstack/localstack:latest"));
```
Comment (eddumelendez) [inline langchain4j-opensearch/src/test/java/dev/langchain4j/store/embedding/opensearch/OpenSearchEmbeddingStoreAWSTest.java:33]: using latest due to a recent fix that it is only available in latest. Later should use a fixed version

**Review by ghost — DISMISSED** [review]
@eddumelendez thank you so much!

**Comment by langchain4j** [comment]
@eddumelendez BTW I am thinking about maintaining a few publicly available docker images with ollama and pre-downloaded models to simplify it for the end user.

This seems to work, but I would appreaciate if you could review it:

```Dockerfile
# Start from the ollama/ollama base image
FROM ollama/ollama:0.1.17

# Override the entrypoint and command to execute the model download
RUN /bin/sh -c "/bin/ollama serve & sleep 1 && ollama pull phi"

# Reset the entrypoint and command to their original values
ENTRYPOINT ["/bin/ollama"]
CMD ["serve"]
```
And then
```
docker build -t ollama-phi:0.1.17 .
docker tag ollama-phi:0.1.17 langchain4j/ollama-phi:0.1.17
docker push langchain4j/ollama-phi:0.1.17
```

I have not much experience with docker, so I am not sure if I am doing it right or if it is a good idea in general.
WHYT? Thanks!

**Comment by langchain4j** [comment]
@eddumelendez as far as I understand I can automate this process with docker pro and just add more or update existing docker files hosted on github (to keep ollama and model versions up-to-date)?

**Comment by langchain4j** [comment]
I have created a few repositories: https://hub.docker.com/repositories/langchain4j

**Comment by eddumelendez** [comment]
@langchain4j sorry for the delay. I have created [this issue](https://github.com/jmorganca/ollama/issues/2160) in Ollama to make the process a little bit more simple. I checked the image and looks like the ARM image is not starting, I'll take a look. I think those Dockerfiles can be moved to another repository and we can automate the process by adding an specific workflow like [this one](https://github.com/testcontainers/moby-ryuk/blob/main/.github/workflows/publish-docker-image.yml). We can chat more about it in another issue.

LMK if there is something else related to this PR.

**Comment by langchain4j** [comment]
@eddumelendez hi! I was fighting with ARM images for quite a while, would appreciate your help with this a lot! Yeah, lets move to a separate repo!

**Comment by langchain4j** [comment]
@eddumelendez I have created https://github.com/langchain4j/langchain4j-docker and added you as collaborator.

I was also thinking about simplifying Ollama IT setup and using one of the modified ollama docker containers (with pre-downloaded models) like this one: https://hub.docker.com/r/langchain4j/ollama-phi
WDYT?

**Comment by eddumelendez** [comment]
> I was also thinking about simplifying Ollama IT setup and using one of the modified ollama docker containers (with pre-downloaded models) like this one: https://hub.docker.com/r/langchain4j/ollama-phi
WDYT?

I think it make sense, will check. Also, I think an official image would be great to have and can be maintained by Ollama itself, so, I have opened https://github.com/jmorganca/ollama/issues/2161. If those images are provided by the official account would reduce the maintenance of those images and the new supported modules 🤞🏽. In the meantime, publishing our own can reduce boilerplate code around the IT.

**Review by ghost — APPROVED** [review]
@eddumelendez great job, thank you!

**Pull request merged**